### ISSUE TITLE
Error updating from GTFS-realtime data sources

### ISSUE THREAD

**Issue by metodi-milchev** [OneBusAway/onebusaway-application-modules#253]
Hi. I am trying to run OnBusAway 2.0.0 on Debian 10. After the deployments following the "Configuration and Deployment Guide for v2.x" I am receiving the following error:

`2020-05-31 18:36:09,371 WARN  [GtfsRealtimeSource.java:850] : Error updating from GTFS-realtime data sources
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:657)
	at java.util.ArrayList.get(ArrayList.java:433)
	at org.onebusaway.transit_data_federation.impl.realtime.gtfs_realtime.GtfsRealtimeSource.handleCombinedUpdates(GtfsRealtime$
	at org.onebusaway.transit_data_federation.impl.realtime.gtfs_realtime.GtfsRealtimeSource.handeUpdates(GtfsRealtimeSource.ja$
	at org.onebusaway.transit_data_federation.impl.realtime.gtfs_realtime.GtfsRealtimeSource.refresh(GtfsRealtimeSource.java:39$
	at org.onebusaway.transit_data_federation.impl.realtime.gtfs_realtime.GtfsRealtimeSource$RefreshTask.run(GtfsRealtimeSource$
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)    
`

Will appreciate any help and suggestion how to make the transit-data-source to update from GTFS-realtime data sources. Thanks!

**Comment by sheldonabrown** [comment]
OneBusAway encountered some data it wasn't expecting.  Likely an empty list of stop time updates, but unfortunately the exception doesn't pinpoint the error.

Is the GTFS-RT feed you are using public?  If so, can you post it here?  And have you verified the GTFS you are using loaded successfully and matches the GTFS-RT feed?

**Comment by metodi-milchev** [comment]
Thanks for your suggestion. I did check for empty stop time updates and excluded them from the feed. The result is the same - IndexOutOfBoundsException.
The same GTFS Static and Realtime data works on the OnaBusAway 1.1.19 QuickStart I tried first on Windows, but I decided to check OBA ver. 2
I am attaching the XML representation of the PB files and the last PB files if this can help. Thank you!
[GTFS-rt-0531.zip](https://github.com/OneBusAway/onebusaway-application-modules/files/4708109/GTFS-rt-0531.zip)
